lib/linear_dispatcher.rb

```diff
@@ -1,21 +1,24 @@
 class LinearDispatcher < Dispatcher
-  def next_approval(cart)
-    cart.ordered_awaiting_approvals.first
+  def next_pending_approval(cart)
+    # we don't care how the cart was approved/rejected
+    if cart.pending?
+      return cart.ordered_awaiting_approvals.first
+    end
```

Comment:
I'm laughing to myself as I type this, but mind making the return value more explicit, but *excluding* the explicit `return`?

```ruby
def next_pending_approval(cart)
  # we don't care how the cart was approved/rejected
  if cart.pending?
    cart.ordered_awaiting_approvals.first
  else
    nil
  end
end
```

#rubyisms